koku/masu/util/aws/common.py

```diff
@@ -473,6 +584,32 @@ def _get_s3_objects(s3_path):
     return s3_resource.Bucket(settings.S3_BUCKET_NAME).objects.filter(Prefix=s3_path)
 
 
+def get_s3_csv_lastest_daily_date(s3_path, start_date, context, request_id):
```

Comment:
It might be better to do a trino query to get the date instead because it gives us what we actually have in the db:
```
select DISTINCT date(lineitem_usagestartdate) as date from hive.org1234567.aws_line_items order by date desc limit 1;
```
Just a thought.

Reply:
YES! I like this idea! And it actually solves another issue I have, better than the way I was solving it :wink:

Reply:
Although actually it only partially solves my issue. so I might need a combination or just stick with what I now have. :thinking:

Reply:
I kept my current solution for now, it does add another tiny migration to the one I was already doing. BUT... This method also means its a NO-OP to deploying this stuff out! Like merge and walk away style, it _should_ now just magically cleanup the old files and create the full month of daily archives when the next download comes in. :+1:

Reply:
I like the self healing approach here 👍 Awesome work Luke!